src/kagglehub/models_helpers.py

```diff
@@ -0,0 +1,227 @@
+    }
+    api_client = KaggleApiV1Client()
+    response = api_client.post("/blobs/upload", data=data)
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    with open(file_path, "rb") as f:
+        file_data = f.read()
+
+    headers = {"Content-Type": "application/octet-stream"}
+    try:
+        requests.put(response["createUrl"], data=file_data, headers=headers, timeout=10)
+    except requests.exceptions.Timeout:
+        # Handle the timeout exception
+        logger.error("Request timed out")
+
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    return response["token"]
+
+
+def upload_files(folder, quiet=False, dir_mode="skip"):  # noqa: FBT002
+    """upload files in a folder
+        Parameters
+    ==========
+    request: the prepared request
+    folder: the folder to upload from
+    quiet: suppress verbose output (default is False)
+    """
+    tokens = []
+    for file_name in os.listdir(folder):
+        if file_name in [
+            DATASET_METADATA_FILE,
+            OLD_DATASET_METADATA_FILE,
+            KERNEL_METADATA_FILE,
+            MODEL_METADATA_FILE,
+            MODEL_INSTANCE_METADATA_FILE,
+        ]:
+            continue
+        tokens.append(_upload_file_or_folder(folder, file_name, dir_mode, quiet))
+    return tokens
+
+
+def _upload_file_or_folder(parent_path, file_or_folder_name, dir_mode, quiet=False):  # noqa: FBT002
+    full_path = os.path.join(parent_path, file_or_folder_name)
+    if os.path.isfile(full_path):
+        return _upload_file(file_or_folder_name, full_path, quiet)
+
+    elif os.path.isdir(full_path):
+        if dir_mode in ["zip", "tar"]:
+            with DirectoryArchive(full_path, dir_mode) as archive:
+                return _upload_file(archive.name, archive.path, quiet)
+        elif not quiet:
+            logger.info("Skipping folder: " + file_or_folder_name + "; use '--dir-mode' to upload folders")
+    elif not quiet:
+        logger.info("Skipping: " + file_or_folder_name)
+    return None
+
+
+def _upload_file(file_name, full_path, quiet):
+    """Helper function to upload a single file
+    Parameters
+    ==========
+    file_name: name of the file to upload
+    full_path: path to the file to upload
+    quiet: suppress verbose output
+    :return: None - upload unsuccessful; instance of UploadFile - upload successful
+    """
+
+    if not quiet:
+        logger.info("Starting upload for file " + file_name)
+
+    content_length = os.path.getsize(full_path)
+    token = _upload_blob(full_path)
+    if token is None:
+        if not quiet:
+            logger.info("Upload unsuccessful: " + file_name)
+        return None
+    if not quiet:
+        logger.info("Upload successful: " + file_name + " (" + File.get_size(content_length) + ")")
+    return token
+
+
+def create_model(owner_slug, model_slug):
```

Comment:
given this method is not used outside this module add the `_` prefix (means "private" by convention).